src/CommunitySleep.js

```diff
@@ -0,0 +1,102 @@
+    const sleepQuality = userSleep.map(sleeper => sleeper.sleepQuality);
+    const avgSleepQuality = (sleepQuality.reduce((a, b) => a + b, 0)) / sleepQuality.length;
+    return Math.round(avgSleepQuality * 10) / 10;
+  }
+
+  findHrsSleptOnDay(userID, date) {
+    const hoursSlept = this.sleeps.find(sleeper => sleeper.userID === userID && sleeper.date === date);
+    return hoursSlept.hoursSlept;
+  }
+
+  findSleepQualityOnDay(userID, date) {
+    const sleepQuality = this.sleeps.find(sleeper => sleeper.userID === userID && sleeper.date === date);
+    return sleepQuality.sleepQuality;
+  }
+
+  calculateSleepStatsWeek(userID, startDate, endDate) {
+    const sleepWeekTotals = [];
+    const startDateNumber = this.convertDateString(startDate);
+    const endDateNumber = this.convertDateString(endDate);
+    const userSleep = this.findUserSleepData(userID);
+    const sleepDates = userSleep.filter(sleeper => {
+      const sleepDateToNumber = this.convertDateString(sleeper.date)
+      if(sleepDateToNumber >= startDateNumber && sleepDateToNumber <= endDateNumber){
+        sleepWeekTotals.push(sleeper);
+      }
+    })
+    return sleepWeekTotals;
+  }
+
+  calculateAvgSleepQualityWk(userID, startDate, endDate) {
+    const something = this.calculateSleepStatsWeek(userID, startDate, endDate);
+    const sleepQualityTotals = something.map(sleeper => sleeper.sleepQuality)
+    const sleepWeekAvg = (sleepQualityTotals.reduce((a, b) => a + b, 0))/7
+    return Math.round(sleepWeekAvg * 10) / 10
+  }
+
+  calculateAvgSleepQuality() {
+    const sleepQualityAllUsers = this.sleeps.map(allSleepers => allSleepers.sleepQuality);
+     const avgSleepQuality = (sleepQualityAllUsers.reduce((a, b) => a + b, 0)) / sleepQualityAllUsers.length;
+     return Math.round(avgSleepQuality * 10) / 10;
+  }
+
+  findBestQualitySleepers(startDate, endDate) {
+    const bestSleepers = []
+    const startDateNumber = this.convertDateString(startDate);
+    const endDateNumber = this.convertDateString(endDate);
+    const sleepWeek = this.sleeps.filter(sleeper => {
+      const sleepDateToNumber = this.convertDateString(sleeper.date)
+      if (sleepDateToNumber >= startDateNumber && sleepDateToNumber <= endDateNumber) {
+        return sleeper;
+      }
+    })
+    const filterSleepers = sleepWeek.filter(sleeper => {
+      const avgSleepQual = this.calculateAvgSleepQualityWk(sleeper.userID, startDate, endDate)
+      if(!bestSleepers.includes(sleeper.userID) && avgSleepQual > 3){
+        bestSleepers.push(sleeper.userID)
+      }
+    })
+    return bestSleepers;
+  }
+  findMostHrsSleepers(date) {
+    const sleepers = [];
+    const sleepHours = [];
+    this.sleeps.forEach(sleeper => {
+      sleepHours.push(this.findHrsSleptOnDay(sleeper.userID, date))
+      const longestSleep = Math.max(...sleepHours);
+      if(longestSleep === sleeper.hoursSlept && date === sleeper.date) {
+        sleepers.push(sleeper.userID);
+      }
+    })
+    return sleepers;
+  }
```

Comment:
@srslie  I'm not sure I follow what you mean.  As far as what I thought I accomplished, I used the method to find how many hours for each user were spent sleeping on a certain date and pushed to an array, and then the new array is assessed through the conditional to meet the requirements of matching the date and hours slept.  You would have to show me where and how you're thinking it could be buggy, since it passed my tests.  Also, the function (according to the spec) is allowed to return more than one user, or are you implying they'd have different values for sleep hours?